File: moduleb/src/main/java/com/braisgabin/dagger/moduleb/ComponentB.java
package com.braisgabin.dagger.moduleb;

import com.braisgabin.dagger.modulec.ComponentC;
import dagger.Component;

@Component(dependencies = ComponentC.class)
public interface ComponentB {

    @Component.Builder
    interface Builder {

        Builder with(ComponentC component);

        ComponentB build();
    }

    String key();
}

File: moduleb/src/main/java/com/braisgabin/dagger/moduleb/ComponentB.kt
package com.braisgabin.dagger.moduleb

import com.braisgabin.dagger.modulec.ComponentC
import dagger.Component

@Component(dependencies = [ComponentC::class])
interface ComponentB {

    @Component.Builder
    interface Builder {

        fun with(component: ComponentC): Builder

        fun build(): ComponentB
    }

    fun key(): String
}
